worlds/cv64/__init__.py

```diff
@@ -0,0 +1,331 @@
+
+    def create_items(self) -> None:
+        item_counts = get_item_counts(self)
+
+        # Set up the items correctly
+        for classification in item_counts:
+            for item in item_counts[classification]:
+                self.multiworld.itempool += [self.create_item(item, classification)
+                                             for _ in range(item_counts[classification][item])]
+
+    def set_rules(self) -> None:
+        active_regions = self.multiworld.get_regions(self.player)
+
+        # Set the required Special2s to the number of available bosses or crystal events returned if Special2s
+        # are not the goal.
+        if self.options.draculas_condition == self.options.draculas_condition.option_crystal:
+            self.total_s2s = 1
+            self.required_s2s = 1
+        elif self.options.draculas_condition != self.options.draculas_condition.option_specials:
+            self.total_s2s = 0
+            self.required_s2s = 0
+
+        for region in active_regions:
+            # Place event items where they should be.
+            for loc in region.locations:
+                if loc.address is None:
+                    event_item = get_location_info(loc.name, "event")
+                    loc.place_locked_item(self.create_item(event_item, "progression"))
+                    # If we're looking at a boss kill trophy, increment the total S2s and, if we're not already at the
+                    # set number of required bosses, the total required number. This way, we can prevent gen failures
+                    # should the player set more bosses required than there are total.
+                    if event_item == iname.trophy:
+                        self.total_s2s += 1
+                        if self.required_s2s < self.options.bosses_required.value:
+                            self.required_s2s += 1
+
+        # Set up the rules properly.
+        CV64Rules(self).set_cv64_rules()
+
+    def pre_fill(self) -> None:
+        # If we need more Special1s to warp out of Sphere 1 than there are locations available, then AP's fill
+        # algorithm may try placing the Special1s anyway despite placing the stage's single key always being an option.
+        # To get around this problem in the fill algorithm, the keys will be forced early in these situations to ensure
+        # the algorithm will pick them over the Special1s.
+        if self.starting_stage == rname.tower_of_science:
+            if self.s1s_per_warp > 3:
+                self.multiworld.local_early_items[self.player][iname.science_key2] = 1
+        elif self.starting_stage == rname.clock_tower:
+            if (self.s1s_per_warp > 2 and not self.options.multi_hit_breakables.value) or \
+                    (self.s1s_per_warp > 8 and self.options.multi_hit_breakables.value):
+                self.multiworld.local_early_items[self.player][iname.clocktower_key1] = 1
+        elif self.starting_stage == rname.castle_wall:
```

Comment:
there's no final else can it be anything other than these 3 or is there potential for it?

Reply:
It's only those three stages being first that can be problematic right now. All the other stages have enough Locations in them to accommodate 10 Special1s if the generator wants to expect the player to warp right away regardless of the settings and/or the end of them being reachable with nothing, at which point I expect subsequent stages to have enough to accommodate the rest.